solr/core/src/test/org/apache/solr/search/TestThinCache.java

```diff
@@ -26,13 +26,63 @@
 import org.apache.solr.SolrTestCaseJ4;
 import org.apache.solr.metrics.SolrMetricManager;
 import org.apache.solr.metrics.SolrMetricsContext;
+import org.apache.solr.util.EmbeddedSolrServerTestRule;
+import org.apache.solr.util.TestHarness;
 import org.apache.solr.util.stats.MetricUtils;
 import org.junit.BeforeClass;
+import org.junit.ClassRule;
 import org.junit.Test;
 
 /** Test for {@link ThinCache}. */
 public class TestThinCache extends SolrTestCaseJ4 {
 
+  @ClassRule public static EmbeddedSolrServerTestRule solrRule = new EmbeddedSolrServerTestRule();
+  public static final String SOLR_NODE_LEVEL_CACHE_XML =
+      "<solr>\n"
+          + "  <caches>\n"
+          + "    <cache name='myNodeLevelCache'\n"
+          + "      size='10'\n"
+          + "      initialSize='10'\n"
+          + "      />\n"
+          + "    <cache name='myNodeLevelCacheThin'\n"
+          + "      class='solr.ThinCache$NodeLevelCache'\n"
+          + "      size='10'\n"
+          + "      initialSize='10'\n"
+          + "      />\n"
+          + "  </caches>\n"
+          + "</solr>";
+
+  @BeforeClass
+  public static void setupSolrHome() throws Exception {
+    Path home = createTempDir("home");
+    Files.writeString(home.resolve("solr.xml"), SOLR_NODE_LEVEL_CACHE_XML);
+
+    solrRule.startSolr(home);
+
+    Path configSet = createTempDir("configSet");
+    copyMinConf(configSet.toFile());
+    // insert a special filterCache configuration
+    Path solrConfig = configSet.resolve("conf/solrconfig.xml");
+    Files.writeString(
+        solrConfig,
+        Files.readString(solrConfig)
+            .replace(
```

Comment:
This is a bit hacky I admit.  I would like to add a utility method that might look something like
 `updateXml(Path, String xpath, String newXml)` where the Xpath would be like "/config/query/filterCache".  ChatGPT wrote one for me :-)

Reply:
When possible, the Config API can be used as is done in `org.apache.solr.search.json.TestJsonRequestWithEdismaxDefType#test`.  But "parentCacheName" isn't supported / is too exotic.  I hate that the Config API is a custom format and only partially implements what we can specify in XML.  I'm torn between embracing it and pretending it doesn't exist.